OpenSkyPlusInjectors/OpenSkyPlusInjectors.cs

```diff
@@ -1,155 +1,155 @@
-        }
-        catch (Exception ex)
-        {
-            throw new Exception($"Failed to apply shot data patch:\n{ex}");
-        }
-
-        _logger.LogDebug("Shot data patch successfully applied");
-        MessageOpenSkyPlusReady?.Invoke();
-    }
-
-    private static void ShotDataPostFixPatch(object IJHAAAHONOE)
-    {
-        MessageNewShot?.Invoke(IJHAAAHONOE);
-    }
+using System;
+using System.Reflection;
+using BepInEx.Logging;
+using HarmonyLib;
+
+namespace OpenSkyPlus;
+
+public delegate void NotificationOpenSkyPlusReady();
+
+public delegate void NotificationNewShot(object launchMonitorShot);
+
+public static class OpenSkyPlusApiInjector
+{
+    private static ManualLogSource _logger;
+
+    private static readonly Harmony Harmony = new("OpenSkyPlus");
+
+    private static bool _deviceReady;
+    public static event NotificationDeviceReady MessageDeviceReady;
+    public static event NotificationOpenSkyPlusReady MessageOpenSkyPlusReady;
+    public static event NotificationNewShot MessageNewShot;
+    public static event NotificationMonitorStatusChange MessageMonitorConnected;
+    public static event NotificationMonitorStatusChange MessageMonitorDisconnected;
+
+    public static void Initialize(ManualLogSource log)
+    {
+        _logger = log;
+        DeviceReadyCheckPatch();
+        ApiSubscriber.MessageApiLoaded += ShotDataInjector;
+        ApiSubscriber.MessageApiLoaded += DeviceConnectedPatch;
+        ApiSubscriber.MessageApiLoaded += DeviceDisconnectedPatch;
+    }
+
+    private static void DeviceReadyCheckPatch()
+    {
+        try
+        {
+            Harmony.Patch(
+                AccessTools.TypeByName("Security.AOMKMFBGCGG")
+                    .GetMethod("OAPNBCHPPBI", BindingFlags.NonPublic | BindingFlags.Instance),
+                postfix: new HarmonyMethod(typeof(OpenSkyPlusApiInjector), nameof(DeviceReadyPostFixPatch)));
+        }
+        catch (Exception ex)
+        {
+            throw new Exception($"Failed to apply device ready patch:\n{ex}");
+        }
+
+        _logger.LogDebug("Device ready check patch successfully applied");
+    }
+
+    private static void DeviceReadyPostFixPatch()
+    {
+        try
+        {
+            if (_deviceReady)
+                return;
+
+            if (
+                !((bool?)AccessTools.TypeByName("Security.ELOOGCNONDM")
+                    .GetProperty("GPCDJOPENGH")?
+                    .GetValue(AccessTools
+                        .TypeByName("Security.AOMKMFBGCGG")
+                        .GetProperty("KGJFDOMIPBN", BindingFlags.Public | BindingFlags.Static)?
+                        .GetValue(null)) ?? false)
+            ) return;
+
+            _deviceReady = true;
+            _logger.LogDebug("Device Ready");
+            MessageDeviceReady?.Invoke(_logger);
+            MessageMonitorConnected?.Invoke();
+            if (DeviceControls.Armed is false)
```

Comment:
same with style mixing

Reply:
About this one, I am not sure.  I would have to check but I made the Armed nullable now(in the event wehere we are able to query it and set its initial value) and this one check for a false value.  A bit tricky but I am not sure the bang '!' is appropriate